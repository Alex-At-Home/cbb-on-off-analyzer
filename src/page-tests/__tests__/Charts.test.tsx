import React from 'react';
import ChartsPage from '../../pages/Charts';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe("ChartsPage", () => {
  test("ChartsPage - should create snapshot", () => {
    const wrapper = shallow(<ChartsPage/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
