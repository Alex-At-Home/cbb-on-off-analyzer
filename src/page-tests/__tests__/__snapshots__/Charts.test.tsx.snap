// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartsPage ChartsPage - should create snapshot 1`] = `
<Container
  fluid={false}
>
  <SiteModeDropdown />
  <Row
    className="mt-2"
    noGutters={false}
  >
    <Col
      className="text-center"
      xs={12}
    >
      <LandingPageIcon />
      <h3>
        Analysis Tool - Interesting Charts
      </h3>
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <HeaderBar
      common={Object {}}
      thisPage="charts"
    />
  </Row>
  <Row
    className="border-top"
    noGutters={false}
  >
    <Nav
      fill={false}
      justify={false}
    >
      <NavDropdown
        id="chartSelection"
        title="View Charts"
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          eventKey="high_major_positions"
          href="#high_major_positions"
          onSelect={[Function]}
        >
          Position Analysis 2010+
        </DropdownItem>
      </NavDropdown>
    </Nav>
  </Row>
  <Row
    noGutters={false}
  >
    <TabContainer
      activeKey="high_major_positions"
    >
      <Row
        noGutters={false}
      >
        <ForwardRef>
          <TabPane
            eventKey="high_major_positions"
          >
            <div>
              <Alert
                className="small"
                closeLabel="Close alert"
                show={true}
                transition={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "appear": false,
                      "in": false,
                      "mountOnEnter": false,
                      "timeout": 300,
                      "unmountOnExit": false,
                    },
                    "render": [Function],
                  }
                }
                variant="info"
              >
                <em>
                  A 2-d visualization of the different positions (P6 2010-2020, all D1 2015-2020).
                  <span>
                     A description of the fields and methodology used is
                     
                    <a
                      href="https://hoop-explorer.blogspot.com/2020/05/classifying-college-basketball.html"
                      target="_blank"
                    >
                      here
                    </a>
                    .
                  </span>
                   The data came from
                   
                  <a
                    href="http://adamcwisports.blogspot.com/p/data.html"
                    target="_blank"
                  >
                    barttorvik.com
                  </a>
                  , much appreciated!
                  <br />
                  The graph below is interactive - you can scroll in or out, move around, highlight specific teams, etc. Note - it can be a bit slow!
                  <br />
                </em>
              </Alert>
              <LoadingOverlay
                active={true}
                text="Loading Chart"
              >
                <iframe
                  frameBorder="0"
                  height="700px"
                  onLoad={[Function]}
                  src="/high_major_positions.html"
                  width="1024px"
                />
              </LoadingOverlay>
              <Alert
                className="small"
                closeLabel="Close alert"
                show={true}
                transition={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "appear": false,
                      "in": false,
                      "mountOnEnter": false,
                      "timeout": 300,
                      "unmountOnExit": false,
                    },
                    "render": [Function],
                  }
                }
                variant="info"
              >
                <em>
                  The x- and y- axis are linear combinations of the input features that are hard to interpret in basketball terms. But basically:
                  <ul>
                    <li>
                      Left-to-right is "guard-like" to "forward-like":
                    </li>
                    <ul>
                      <li>
                        leftwards: 3P%, Assist rate, 3P rate, steal rate
                      </li>
                      <li>
                        rightwards: DRB%, ORB%, dunk rate, mid-2P rate, block rate, foul rate
                      </li>
                    </ul>
                    <li>
                      Top-to-bottom is a mishmash of position-specific bonuses/penalties: combined because we only have 2-d to play with!
                    </li>
                    <ul>
                      <li>
                        upwards: AST/TOV, Assist rate, TO rate, steal rate; dunk/rim rate, mid-2P rate, block rate
                      </li>
                      <li>
                        downwards: 3P rate, rim shooting ability, assists/FG; DRB%, ORB%
                      </li>
                    </ul>
                  </ul>
                </em>
              </Alert>
            </div>
          </TabPane>
        </ForwardRef>
      </Row>
    </TabContainer>
  </Row>
</Container>
`;
